### PR TITLE
No more public underscored methods

### LINKED ISSUE #1917: Remove "package-protected" fields
We have a number of fields in our classes that look like this:
```Typescript
public _packageProtectedField;
```
We should make them fully `public`, `protected`, or `private. This may be tricky for some things, such as the layout code -- those will receive separate subissues.

### PR DESCRIPTION
**Pull request by aicioara**
Closes #1917

Except for Drawers, which are going to be removed anyway.

As part of this
- Some fields were actually private
- Some fields were actually protected
- <del> An entire test class was removed as it was not referenced anyway and functionality was actually removed, according to @jtlan </del> The drawers will be refactored in #1913, so not touching that now

### CODE REVIEW

**Comment by aicioara** [comment]
No QE, no `.js` file was modified

**Comment by rcchen** [comment]
I think you're missing a couple of functions, specifically in all of the Drawers.

This regular expression can find all of the public underscore methods: `^\s*_[\w]+\(`

**Comment by aicioara** [comment]
The drawers refactor is part of the bigger #1913 . Won't fix here

File: test/interactions/hoverInteractionTests.ts [hunk 1]
```diff
@@ -1,222 +0,0 @@
-///<reference path="../testReference.ts" />
-
-var assert = chai.assert;
-
-class TestHoverable extends Plottable.Component.AbstractComponent implements Plottable.Interaction.Hoverable {
```
Comment (rcchen) [inline test/interactions/hoverInteractionTests.ts]: Why were these tests deleted?

Comment (aicioara) [inline test/interactions/hoverInteractionTests.ts]: They should have been removed when the hoverInteraction disappeared

File: test/scales/modifiedLogScaleTests.ts [hunk 2]
```diff
@@ -91,7 +91,7 @@ describe("Scales", () => {
       scale.domainMin(minInMiddle);
       var requestedDomain2 = [-10, 10];
       scale.addExtentsProvider((scale: Plottable.Scales.ModifiedLog) => [requestedDomain2]);
-      scale._autoDomainIfAutomaticMode();
+      (<any>scale)._autoDomainIfAutomaticMode();
```
Comment (rcchen) [inline test/scales/modifiedLogScaleTests.ts]: Why is this `<any>` casted?

File: test/scales/modifiedLogScaleTests.ts [hunk 3]
```diff
@@ -120,7 +120,7 @@ describe("Scales", () => {
       scale.domainMax(maxInMiddle);
       var requestedDomain2 = [-10, 10];
       scale.addExtentsProvider((scale: Plottable.Scales.ModifiedLog) => [requestedDomain2]);
-      scale._autoDomainIfAutomaticMode();
+      (<any>scale)._autoDomainIfAutomaticMode();
```
Comment (rcchen) [inline test/scales/modifiedLogScaleTests.ts]: Again.

File: test/scales/timeScaleTests.ts [hunk 4]
```diff
@@ -77,7 +77,7 @@ describe("TimeScale tests", () => {
     scale.domainMin(minInMiddle);
     var requestedDomain2 = [new Date("2014-05-01"), new Date("2016-07-01")];
     scale.addExtentsProvider((scale: Plottable.Scales.Time) => [requestedDomain2]);
-    scale._autoDomainIfAutomaticMode();
+    (<any>scale)._autoDomainIfAutomaticMode();
```
Comment (rcchen) [inline test/scales/timeScaleTests.ts]: Again.

File: test/scales/timeScaleTests.ts [hunk 5]
```diff
@@ -112,7 +112,7 @@ describe("TimeScale tests", () => {
     scale.domainMax(maxInMiddle);
     var requestedDomain2 = [new Date("2014-05-01"), new Date("2016-07-01")];
     scale.addExtentsProvider((scale: Plottable.Scales.Time) => [requestedDomain2]);
-    scale._autoDomainIfAutomaticMode();
+    (<any>scale)._autoDomainIfAutomaticMode();
```
Comment (rcchen) [inline test/scales/timeScaleTests.ts]: Again.

File: test/scales/modifiedLogScaleTests.ts (hunk 2 above)
Comment (aicioara) [inline test/scales/modifiedLogScaleTests.ts]: Protected method call

File: test/scales/modifiedLogScaleTests.ts (hunk 3 above)
Comment (aicioara) [inline test/scales/modifiedLogScaleTests.ts]: same

File: test/scales/timeScaleTests.ts (hunk 4 above)
Comment (aicioara) [inline test/scales/timeScaleTests.ts]: same

File: test/scales/timeScaleTests.ts (hunk 5 above)
Comment (aicioara) [inline test/scales/timeScaleTests.ts]: same

File: test/interactions/hoverInteractionTests.ts (hunk 1 above)
Comment (aicioara) [inline test/interactions/hoverInteractionTests.ts]: Also, the functionality is now tested in the `pointerInteractionTests`

**Comment by bluong** [comment]
I think @jtlan is attacking some piece of the Drawer public underscored methods through the usage in `getAllPlotData()` etc. and I'm attacking the rest of the pieces in a more overarching `Drawer` refactor.  So, don't worry about those I think

File: test/scales/modifiedLogScaleTests.ts (hunk 2 above)
Comment (bluong) [inline test/scales/modifiedLogScaleTests.ts]: Does it make sense to call `scale.autoDomain()` instead?

File: test/scales/modifiedLogScaleTests.ts (hunk 3 above)
Comment (bluong) [inline test/scales/modifiedLogScaleTests.ts]: Does it make sense to call `scale.autoDomain()` instead?

File: test/scales/timeScaleTests.ts (hunk 4 above)
Comment (bluong) [inline test/scales/timeScaleTests.ts]: Does it make sense to call `scale.autoDomain()` instead?

File: test/scales/timeScaleTests.ts (hunk 5 above)
Comment (bluong) [inline test/scales/timeScaleTests.ts]: Does it make sense to call `scale.autoDomain()` instead?

File: test/scales/modifiedLogScaleTests.ts (hunk 2 above)
Comment (aicioara) [inline test/scales/modifiedLogScaleTests.ts]: I wanted to keep the original behavior

Comment (bluong) [inline test/scales/modifiedLogScaleTests.ts]: Unless I'm mistaken `scale.addExtentsProvider` actually autoDomains by default, I don't think this line is necessary

**Comment by rcchen** [comment]
Please update https://github.com/palantir/plottable/wiki/Changes-from-1.0.0-RC2-to-1.0.0-RC3 after this merges in.

File: src/core/renderPolicy.ts [hunk 6]
```diff
@@ -35,7 +35,7 @@ module Plottable {
      * it.
      */
     export class Timeout implements RenderPolicy {
-      public _timeoutMsec: number = Utils.DOM.POLYFILL_TIMEOUT_MSEC;
+      private _timeoutMsec: number = Utils.DOM.POLYFILL_TIMEOUT_MSEC;
```
Comment (aicioara) [inline src/core/renderPolicy.ts]: do we want to have a setter for this field?

Comment (jtlan) [inline src/core/renderPolicy.ts]: Not critical I think. I kind of doubt that anyone is using this class anyway.

Comment (aicioara) [inline src/core/renderPolicy.ts]: Agree.

Comment (bluong) [inline src/core/renderPolicy.ts]: Out of scope, but should we delete this class?  Let's not expose things unless we know they are used

**Comment by aicioara** [comment]
Updated the wiki page as well

**Pull request merged**